mcache/tests/conftest.py

```diff
@@ -0,0 +1,138 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import subprocess
+import os
+import tempfile
+import shutil
+import time
+import pytest
+import bmemcached
+from bmemcached.exceptions import MemcachedException
+
+from datadog_checks.utils.platform import Platform
+from datadog_checks.mcache import Memcache
+
+from common import (HERE, PORT, HOST, USERNAME, PASSWORD, DOCKER_SOCKET_DIR, DOCKER_SOCKET_PATH)
+
+HOST_SOCKET_DIR = os.path.realpath(tempfile.mkdtemp())
+HOST_SOCKET_PATH = os.path.join(HOST_SOCKET_DIR, 'memcached.sock')
```

Comment:
Let's put these in the `memcached_socket` fixture (uncomment that bit) and use `try` around anything that can fail and put the cleanup in `finally`.

Reply:
Oh actually I found this approach and will try it here https://github.com/DataDog/integrations-core/blob/master/directory/tests/test_directory.py#L91-L114